File: ios/ExpofpModule.m

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(ExpofpModule, NSObject)
RCT_EXTERN_METHOD(preload:(NSString *)url resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
@end

File: ios/ExpofpModule.swift
import ExpoFpFplan
import ExpoFpCommon
import Foundation

@objc(ExpofpModule)
class ExpofpModule: NSObject {

  @objc static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc func preload(_ url: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    SharedFplanView.preload(url, settings: Settings())
    resolve(nil)
  }
}